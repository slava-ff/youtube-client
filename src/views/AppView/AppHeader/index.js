/* eslint no-use-before-define: 2 */
import AppHeader from './header';

import './header-style.css';

export default AppHeader;
