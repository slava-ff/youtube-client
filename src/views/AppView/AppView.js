export default class AppView {
  constructor(clipInfo) {
    this.clipInfo = clipInfo;
  }

  render() {
    const content = document.createElement('ul');
    // eslint-disable-next-line no-plusplus
    for (let i = 0; i < Object.keys(this.clipInfo).length; i++) {
      const li = document.createElement('li');
      let { name } = this.clipInfo[i];
      if (this.clipInfo[i].name.length > 40) {
        name = name.substr(0, 40).concat('...');
      }
      li.innerHTML = `
        <div class="preview" id="preview${i}"><a href="https://www.youtube.com/watch?v=${this.clipInfo[i].id}" target="_blank"><img src="${this.clipInfo[i].preview}" alt="${this.clipInfo[i].preview}"></a></div>
        <div class="name" id="name${i}"><div class="nameContent"><a href="https://www.youtube.com/watch?v=${this.clipInfo[i].id}" target="_blank">${name}</a></div></div>
        <div class="author" id="author${i}">${this.clipInfo[i].author}</div>
        <div class="date" id="date${i}">${this.clipInfo[i].date.substr(0, 10)}</div>
        <div class="views" id="views${i}">${this.clipInfo[i].views} views</div>
        <div class="description" id="description${i}">${this.clipInfo[i].description}</div>
      `;
      content.appendChild(li);
    }
    document.body.appendChild(content);
  }
}
