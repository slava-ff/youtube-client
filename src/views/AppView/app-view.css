@import url('https://fonts.googleapis.com/css?family=Titillium+Web:400,400i,700,700i&subset=latin-ext');

body {
  margin: default;
  background-color: rgb(231, 231, 231);
  font-family: 'Titillium Web', sans-serif;
}

ul {
  list-style: none;
  display: grid;
  grid-template-columns: 22% 22% 22% 22%;
  grid-gap: 4%;
  margin: 0%;
  padding: 1%;
}

ul li {
  margin-top: 20px;
  background-color: white;
  display: flex;
  overflow: hidden;
  flex-direction: column;
  border: 1px solid rgb(180, 180, 180);
  border-radius: 3%;
  height: 365px;
}

ul li:hover {
  background-color: red;
  color: white;
  transition: all 0.2s;
  transform: scale(1.05);
}

ul li .preview {
  position: relative;
  top: 3%;
}

ul li .preview img {
  width: 100%;
}

ul li .name {
  z-index: 2;
  position: relative;
  bottom: 30%;
}

ul li .name .nameContent {
  background-color: yellow;
  padding: 3% 2%;
  text-align: center;
  justify-content: center;
  font-size: 1.15em;
  font-weight: bold;
  line-height: 1.2em;
  height: 2.4em;
  vertical-align: middle;
  /* min-height: 2.4em; */
}

ul li .author {
  padding: 0.5% 0%;
  font-size: 0.9em;
  text-align: center;
  font-weight: bold;
  position: relative;
  bottom: 14%;
}

ul li .date {
  padding: 0.5% 0%;
  font-size: 0.9em;
  text-align: center;
  font-weight: bold;
  position: relative;
  bottom: 14%;
}

ul li .views {
  padding: 0.5% 0%;
  font-size: 0.9em;
  text-align: center;
  font-weight: bold;
  position: relative;
  bottom: 14%;
}

ul li .description {
  padding: 0.5% 2%;
  font-size: 0.8em;
  position: relative;
  bottom: 14%;
}
