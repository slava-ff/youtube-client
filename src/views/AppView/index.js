/* eslint no-use-before-define: 2 */
import AppView from './AppView';

import './app-view.css';

export default AppView;
