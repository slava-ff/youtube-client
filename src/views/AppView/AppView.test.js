import AppView from './AppView';

describe('AppView.prototype.render', () => {
  it('Should be an instance of Fucntion', () => {
    expect(AppView.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render correctly', () => {
    const context = {
      titles: [
        '123',
        '1234',
        '12345',
      ],
    };

    AppView.prototype.render.call(context);

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
