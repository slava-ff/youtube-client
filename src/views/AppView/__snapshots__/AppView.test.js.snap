// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.prototype.render Should be render correctly 1`] = `"<ul><li>123</li><li>1234</li><li>12345</li></ul>"`;
